Buttplug.Server.Managers.ErosTek/ET312Device.cs

```diff
@@ -0,0 +1,398 @@
+﻿using System;
+using System.IO.Ports;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+using Buttplug.Core;
+using Buttplug.Core.Messages;
+
+namespace Buttplug.Server.Managers.ETSerialManager
+{
+    public class ET312Device : ButtplugDevice
+    {
+        private SerialPort serialPort;
+        private byte boxkey;
+        private double speed;
+        private double position;
+        private double currentPosition;
+        private double increment;
+        private double fade;
+        private double updateInterval;
+        private Timer updateTimer;
+        private object movementLock;
+
+        public ET312Device(SerialPort port, IButtplugLogManager aLogManager, string name, string id)
+            : base(aLogManager, name, id)
+        {
+            movementLock = new object();
+
+            // Handshake with the box
+            serialPort = port;
+            boxkey = 0;
+            Synch();
+
+            try
+            {
+                // Setup box for remote control
+                Execute(0x00);               // load default routine
+                Poke(0x4083, 0x00);          // disable front panel switches
+                Poke(0x409a, 0x00);          // Channel A: Gate Off
+                Poke(0x419a, 0x00);          // Channel B: Gate Off
+                Poke(0x40ac, 0x00);          // Channel A: Set intensity to static
+                Poke(0x41ac, 0x00);          // Channel B: Set intensity to static
+                Poke(0x40a5, 0x00);          // Channel A: Set intensity value
+                Poke(0x41a5, 0x00);          // Channel B: Set intensity value
+                Poke(0x409c, 0xff);          // stop volume ramp
+                Poke(0x40b5, 0x08);          // Channel A: MA knob sets frequency
+                Poke(0x41b5, 0x08);          // Channel B: MA knob sets frequency
+                Poke(0x40be, 0x04);          // Channel A: Set width from advanced menu
+                Poke(0x41be, 0x04);          // Channel B: Set width from advanced menu
```

Comment:
Magic numbers are better replaced with `const` values. 
For example, creating a enum to hold these number means you can replace it with human readable words and reduce looking up what the magic numbers mean.

e.g. 
```C#
public enum ET312Command
{
    Disable = 0x4083,
    ChannelA = 0x409a,
    // ...
}
```

Then in your constructor you can reference them like so:
```C#
Poke(ET312Command.Disable, 0x00);          // disable front panel switches
Poke(ET312Command.ChannelA, 0x00);          // Channel A: Gate Off
// ...
```